File: 2.0/java/texts/findTextsPharmacyReminders.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.callstexts.model.Text;
import com.callfire.api.client.api.callstexts.model.request.FindTextsRequest;
import com.callfire.api.client.api.common.model.Page;

import java.util.Calendar;
import java.util.GregorianCalendar;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api_login", "api_password");
        FindTextsRequest request = FindTextsRequest.create()
            .campaignId(13L)
            .label("reminders")
            .intervalBegin(new GregorianCalendar(2015, Calendar.DECEMBER, 1, 0, 0, 0).getTime())
            .intervalEnd(new GregorianCalendar(2015, Calendar.DECEMBER, 10, 0, 0, 0).getTime())
            .limit(10L)
            .build();
        Page<Text> texts = client.textsApi().find(request);
    }
}

File: 2.0/csharp/texts/findTextsPharmacyReminders.cs
using System;
using CallfireApiClient;
using CallfireApiClient.Api.CallsTexts.Model;
using CallfireApiClient.Api.CallsTexts.Model.Request;
using CallfireApiClient.Api.Common.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var request = new FindTextsRequest
        {
            // find all texts which belong to a particular campaign and were sent between
            // 2015-12-01 and 2015-12-10
            CampaignId = 13,
            Label = "reminders",
            IntervalBegin = new DateTime(2016, 12, 1, 0, 0, 0),
            IntervalEnd = new DateTime(2016, 12, 1, 0, 0, 0),
            Limit = 10
        };
        Page<Text> texts = client.TextsApi.Find(request);
    }
}
